kasa/tapo/tapoplug.py

```diff
@@ -0,0 +1,72 @@
+"""Module for a TAPO Plug."""
+import logging
+from datetime import datetime, timedelta
+from typing import Any, Dict, Optional, cast
+
+from ..emeterstatus import EmeterStatus
+from ..smartdevice import DeviceType
+from .tapodevice import TapoDevice
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class TapoPlug(TapoDevice):
+    """Class to represent a TAPO Plug."""
+
+    def __init__(
+        self,
+        host: str,
+        *,
+        port: Optional[int] = None,
+        credentials,
+        timeout: Optional[int] = None,
+    ) -> None:
+        super().__init__(host, port=port, credentials=credentials, timeout=timeout)
+        self._device_type = DeviceType.Plug
+
+    async def update(self, update_children: bool = True):
+        """Call the device endpoint and update the device data."""
+        await super().update(update_children)
+
+        self._energy = await self.protocol.query("get_energy_usage")
+        self._emeter = await self.protocol.query("get_current_power")
```

Comment:
The bulb seems to support these commands, too, but I'm not sure if we can detect the availability using `component_nego` or some other way.

Here's the response for the from the bulb (`kasa -d --type tapobulb --username $KASA_TAPO_EMAIL --password $KASA_TAPO_PASSWORD  --host 192.168.xx.xx raw-command "" component_nego`):
```
<< {'component_list': [{'id': 'device', 'ver_code': 2},
                    {'id': 'firmware', 'ver_code': 2},
                    {'id': 'quick_setup', 'ver_code': 3},
                    {'id': 'inherit', 'ver_code': 1},
                    {'id': 'time', 'ver_code': 1},
                    {'id': 'wireless', 'ver_code': 1},
                    {'id': 'schedule', 'ver_code': 2},
                    {'id': 'countdown', 'ver_code': 2},
                    {'id': 'antitheft', 'ver_code': 1},
                    {'id': 'account', 'ver_code': 1},
                    {'id': 'synchronize', 'ver_code': 1},
                    {'id': 'sunrise_sunset', 'ver_code': 1},
                    {'id': 'cloud_connect', 'ver_code': 1},
                    {'id': 'default_states', 'ver_code': 1},
                    {'id': 'preset', 'ver_code': 1},
                    {'id': 'brightness', 'ver_code': 1},
                    {'id': 'color', 'ver_code': 1},
                    {'id': 'color_temperature', 'ver_code': 1},
                    {'id': 'auto_light', 'ver_code': 1},
                    {'id': 'on_off_gradually', 'ver_code': 1},
                    {'id': 'device_local_time', 'ver_code': 1},
                    {'id': 'light_effect', 'ver_code': 1},
                    {'id': 'iot_cloud', 'ver_code': 1},
                    {'id': 'bulb_quick_control', 'ver_code': 1}]}
{"component_list": }
```

I updated the list to the draft bulb PR, so it's easily available.